app/models/metric/helper.rb

```diff
@@ -132,11 +132,11 @@ def self.remove_duplicate_timestamps(recs)
         _log.warn("Multiple rows found for the same timestamp: [#{rec.timestamp}], ids: [#{rec.id}, #{last_rec.id}], resource and id: [#{rec.resource_type}:#{rec.resource_id}]")
 
         # Merge records with the same timestamp
-        last_rec.attributes.each { |a| last_rec[a] ||= rec[a] }
+        last_rec.attribute_names.each { |a| last_rec[a] ||= rec[a] }
       else
         ret << rec
+        last_rec = rec
```

Comment:
I was looking on it with @himdel and it is working because `rec` is object and his reference is passed to `last_rec`. And rec is moved into `ret` so then when I am changing `last_rec` I am still changing element of `ret`. is this explanation enough or should I rewrite it ?